### PR TITLE
Utils script to check the percentile climate outcome

### PR DESCRIPTION
**Pull request by gabriel-abrahao**
## Purpose of this PR

This adds a completely standalone utils script that can be used to verify if the MAGICC7 config used in-GAMS is leading to the intended part of the uncertainty range. It compares `pm_globalMeanTemperature` in the GDXs with the percentiles derived from the `MAGICC7_AR6.R` output script, which must have been run before. 

It's meant to be run directly as it's quite lightweight, and generates figures in `check_output_figures.pdf` and a table in
 `check_output_summary.csv` in the main folder. Instructions are on the script's header.

## Type of change

- [x] New feature

### CODE REVIEW

File: scripts/utils/climate_assessment/check_outcomes.R [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# The scenarios to be checked. Needs the full folder names in the output folder,
+# including the timestamp if there is one
+checkscenarios <- c(
+    "C_o_2c_KLW_d50_newmagicc-rem-6",
+    "C_o_cba_2c_KLW_d5-rem-6",
+    "C_o_cba_2c_KLW_d50-rem-6"
+    # "C_o_cba_2c_KLW_d95-rem-6"
+)
+
+runfolders <- file.path("output",checkscenarios)
```
[the remark this answers is not in the document]
Comment (orichters) [reply scripts/utils/climate_assessment/check_outcomes.R]: you still need to move the file to `scripts/output/comparison/checkClimatePercentiles.R` for it to work…

**Comment by gabriel-abrahao** [comment]
Great suggestions Oli, thanks! I did that, fixed a little bug in the variable names (I get pretty confused with the standards in terms of `outputdir` and `outputdirs` sometimes) and made a little more verbose. Tested it and looks good to merge, but it might be good for @fpiontek to be aware that this is now an output script.

File: scripts/output/comparison/checkClimatePercentiles.R [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+cat("===================== STARTING checkClimatePercentiles.R", "\n")
+cat("===================== see output for final location of output files", "\n")
+if(! exists("source_include")) {
+  # Define arguments that can be read from command line
+  outputdir <- "."
+  flags <- lucode2::readArgs("outputdir", .flags = c(i = "--interactive"))
+} else {
+  outputdir <- outputdirs
+}
```
Comment (orichters) [inline scripts/output/comparison/checkClimatePercentiles.R:36]:
```suggestion
  lucode2::readArgs("outputdirs")
```
Sorry, I think rather I messed that up! I don't think you need more than that

Comment (gabriel-abrahao) [reply scripts/output/comparison/checkClimatePercentiles.R]: Uuuh, elegant, thanks, worked like a charm!

**Comment by gabriel-abrahao** [comment]
@pascal-sauer @LaviniaBaumstark could someone please take a look on this CI check failing? Looks like it's having trouble with an R library, cairo-related it seems, so it's way out of my league. What's super odd is that I'm not adding any new dependencies. Thanks!

**Comment by fpiontek** [comment]
Thanks Gabriel, I looked at it but couldn't provide as useful comments as Oli :-)

**Comment by tonnrueter** [comment]
> @pascal-sauer @LaviniaBaumstark could someone please take a look on this CI check failing? Looks like it's having trouble with an R library, cairo-related it seems, so it's way out of my league. What's super odd is that I'm not adding any new dependencies. Thanks!

The CI jobs have a tendency to randomly fail 🤷 I just restarted the check

File: scripts/output/comparison/checkClimatePercentiles.R [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+filelist <- list.files(runfolders, pattern = ".mif", full.names = TRUE) 
+miflist <- filelist[str_detect(filelist, "REMIND_generic_.*")&!str_detect(filelist, "withoutPlus")]
+
+mifdata <- read.quitte(miflist)
+miftempdata <- mifdata %>% 
+    filter(region == "World", variable %in% c(
+        "MAGICC7 AR6|Surface Temperature (GSAT)|10.0th Percentile",
+        "MAGICC7 AR6|Surface Temperature (GSAT)|33.0th Percentile",
+        "MAGICC7 AR6|Surface Temperature (GSAT)|50.0th Percentile",
+        "MAGICC7 AR6|Surface Temperature (GSAT)|66.0th Percentile",
```
Comment (orichters) [inline scripts/output/comparison/checkClimatePercentiles.R:74]: Maybe a minor issue, but why are you using 66.0th Percentile? At least in NGFS, we always use 33.0th and 67.0th.

**Review by orichters — APPROVED** [review]
Fine from my side assuming it works. Can you send me via private message a link to one of these files? Thanks!

File: scripts/output/comparison/checkClimatePercentiles.R (hunk 3 above)
Comment (gabriel-abrahao) [reply scripts/output/comparison/checkClimatePercentiles.R:65]: Good point, that was an oversight

File: scripts/output/comparison/checkClimatePercentiles.R [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+# an RCP (hence the nomenclature of the config files e.g. RCP20_50.json will be close to the 50th percentile outcome in 
+# a RCP2.0-like scenario). This quantile outcome is of course not guaranteed, so this script helps check the validity of
+# the assumed quantile against the 600-member MAGICC7 AR6 assessment that is run by default at the end of each run. 
+# Therefore, for it to work, the output script MAGICC7_AR6.R must have already run and added the `MAGICC7 AR6|**` 
+# variables to `REMIND_generic_*.mif`. To verify the output, check the figures in `check_output_figures.pdf` and the 
+# table in `check_output_summary.csv`
+#
+require(purrr)
+require(stringr)
+require(quitte)
```
Comment (gabriel-abrahao) [inline scripts/output/comparison/checkClimatePercentiles.R:22]: We're using too much of the tidyverse here, so in this particular case I think it's a lot simpler to load it all. I think the only one not being used here is `lubridate`.

**Pull request merged**